core/kits/documents/tabs/settings-layout.php

```diff
@@ -200,9 +199,71 @@ protected function register_tab_controls() {
 		$this->end_controls_section();
 	}
 
+	/**
+	 * Before Save
+	 *
+	 * Runs Before the Kit document is saved.
+	 *
+	 * For backwards compatibility, when the mobile and tablet breakpoints are updated, we also update the
+	 * old breakpoint settings ('viewport_md', 'viewport_lg' ) with the saved values + 1px. The reason 1px
+	 * is added is because the old breakpoints system was min-width based, and the new system introduced in
+	 * Elementor v3.2.0 is max-width based.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param array $data
+	 * @return array $data
+	 */
+	public function before_save( $data ) {
+		// When creating a default kit, $data['settings'] is empty and should remain empty, so settings.
+		if ( empty( $data['settings'] ) ) {
+			return $data;
+		}
+
+		$prefix = Breakpoints_Manager::BREAKPOINT_OPTION_PREFIX;
+		$mobile_breakpoint_key = $prefix . Breakpoints_Manager::BREAKPOINT_KEY_MOBILE;
+		$tablet_breakpoint_key = $prefix . Breakpoints_Manager::BREAKPOINT_KEY_TABLET;
+
+		/** @var Breakpoint[] $default_breakpoint_config */
+		$default_breakpoint_config = Breakpoints_Manager::get_default_config();
+
+		// Update the old mobile breakpoint. If the setting is empty, use the default value.
+		$data['settings'][ $prefix . 'md' ] = empty( $data['settings'][ $mobile_breakpoint_key ] )
+			? $default_breakpoint_config[ Breakpoints_Manager::BREAKPOINT_KEY_MOBILE ]['default_value'] + 1
+			: $data['settings'][ $mobile_breakpoint_key ] + 1;
+
+		// Update the old tablet breakpoint. If the setting is empty, use the default value.
+		$data['settings'][ $prefix . 'lg' ] = empty( $data['settings'][ $tablet_breakpoint_key ] )
+			? $default_breakpoint_config[ Breakpoints_Manager::BREAKPOINT_KEY_TABLET ]['default_value'] + 1
+			: $data['settings'][ $tablet_breakpoint_key ] + 1;
+
+		return $data;
+	}
+
+	public function on_save( $data ) {
+		if ( ! isset( $data['settings'] ) || Document::STATUS_PUBLISH !== $data['settings']['post_status'] ) {
+			return;
+		}
+
+		$should_compile_css = false;
+
+		/** @var Breakpoint[] $breakpoints_default_config */
+		$breakpoints_default_config = Breakpoints_Manager::get_default_config();
+
+		foreach ( $breakpoints_default_config as $breakpoint_key => $default_config ) {
+			$breakpoint_setting_key = Breakpoints_Manager::BREAKPOINT_OPTION_PREFIX . $breakpoint_key;
+
+			if ( isset( $data['settings'][ $breakpoint_setting_key ] ) ) {
+				$should_compile_css = true;
+			}
+		}
+
```

Comment:
Can't you check if `has_custom_breakpoints()` instead?

Reply:
No because has_custom_breakpoints is based on the existing instance of the breakpoints manager (`Plugin::$instance->breakpoint`). The instance is not updated with control value changes when this method runs.